prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2638,3 +2645,131 @@ void rxr_ep_record_tx_op_completed(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_
 	}
 }
 
+#if EFA_UNIT_TEST
+
+#include "efa_unit_tests.h"
+#include "efa_unit_test_utils.h"
+
+struct rdm_ep_poll_ibv_cq_ex_state {
+	struct efa_cq *efa_cq;
+	struct efa_ep *efa_ep;
+	struct rxr_ep *rxr_ep;
+};
+
+static int setup_rdm_ep_poll_ibv_cq_ex(struct rdm_ep_poll_ibv_cq_ex_state *state)
+{
+	int err;
+	static struct efa_cq *efa_cq;
+	static struct efa_ep *efa_ep;
+	static struct rxr_ep *rxr_ep;
+
+	efa_cq = calloc(1, sizeof(struct efa_cq));
+	efa_ep = calloc(1, sizeof(struct efa_ep));
+	rxr_ep = calloc(1, sizeof(struct rxr_ep));
+
+	if (!efa_cq || !efa_ep)
+		return 1;
+
+	err = efa_cq_initialize(efa_cq);
+	if (err)
+		return 1;
+
+	efa_cq->ibv_cq_ex->start_poll = &_start_poll;
+	efa_cq->ibv_cq_ex->next_poll = &_next_poll;
+	efa_cq->ibv_cq_ex->end_poll = &_end_poll;
+	efa_cq->ibv_cq_ex->read_opcode = &_read_send_code;
+
+	err = efa_ep_initialize(efa_ep);
+	if (err)
+		return 1;
+
+	err = rxr_ep_initialize(rxr_ep, efa_cq, efa_ep);
+	if (err)
+		return 1;
+
+	rxr_ep->util_ep.eq->eq_fid.ops->write = &_fid_eq_write_successful;
+
+	state->efa_cq = efa_cq;
+	state->efa_ep = efa_ep;
+	state->rxr_ep = rxr_ep;
+
+	return 0;
+}
+
+static int teardown_rdm_ep_poll_ibv_cq_ex(struct rdm_ep_poll_ibv_cq_ex_state *state)
+{
+	teardown_efa_cq(state->efa_cq);
+	free(state->efa_cq);
+	free(state->efa_ep);
+	free(state->rxr_ep);
+}
+
+void __wrap_rxr_pkt_handle_send_completion(struct rxr_ep *ep,
+										   struct rxr_pkt_entry *pkt_entry)
+{
+	return mock();
```

Comment:
Do you need to return in a "void" function?

Reply:
Nope `mock()` is sufficient.